### PR TITLE
Update csv script #877

### PR DESCRIPTION
**Pull request by rigazilla**
adding script and a make rule `make update-csv` to update csv file. Only permissions and clusterpermissions are updated atm.
adding the rule as dependency for `make codegen`

### CODE REVIEW

**Comment by dmvolod** [comment]
changed go.mod and go.sum files for me and then fails
```
./build/update-csv.sh: line 7: yq: command not found
./build/update-csv.sh: line 33: yq: command not found
make: *** [Makefile:36: update-csv] Error 127
```

**Comment by rigazilla** [comment]
let me try with `go install` it should take care of the hw arch.
`go env GOPATH` needs to be in the path

**Comment by dmvolod** [comment]
> `go env GOPATH` needs to be in the path

No, the same result. Seems to be `$GOPATH/bin` needs in the PATH, not whole `$GOPATH`. But I'm never used this config before

I think it's possible to automate such kind of restrictions with bash script
go.* files were updated anymore as well as infinispan-operator.clusterserviceversion.yaml - seems to need to be regenerated in this PR

File: build/update-csv.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+YQ_MAJ_VERSION=${1:-4}
+installYQ() {
+    printf "Installing yq ..."
+    GO111MODULE=on go get github.com/mikefarah/yq/v4
```
Comment (dmvolod) [inline build/update-csv.sh:6]: Can we bind to the specific enough for us `yq` version to prevent updates on some codegen steps

Comment (rigazilla) [reply build/update-csv.sh:6]: unfortunately not, only major rel can be specified

Comment (dmvolod) [reply build/update-csv.sh:6]: Not a good news, than we need go away from project folder temporary, i.e. /tmp install yq and than return back

**Comment by rigazilla** [comment]
fixed the PATH setup, csv is modified because indentation is fixed
still looking into the go.mod issue

File: build/update-csv.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    PATH=$GOPATH/bin:$PATH
```
Comment (dmvolod) [inline build/update-csv.sh:7]: It would be nice to set `GOPATH` if it not defined for example to `$HOME\go`

Comment (rigazilla) [reply build/update-csv.sh]: oh right, my idea was to use `go env` but I forgot to that. fixed

File: build/update-csv.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+YQ_MAJ_VERSION=${1:-4}
+installYQ() {
+    printf "Installing yq ..."
+    GO111MODULE=on go get github.com/mikefarah/yq/v4
+    PATH=$(go env GOPATH)/bin:$PATH
```
Comment (dmvolod) [inline build/update-csv.sh:7]: This need not only for installation, usage needs this also

**Comment by ryanemerson** [comment]
I get the following when trying to execute `make codegen`:

```
./build/update-csv.sh
Validating yq installation...Not found
Installing yq ..../build/update-csv.sh: line 7: gobin: command not found
./build/update-csv.sh: line 9: yq: command not found
./build/update-csv.sh: line 36: yq: command not found
make: *** [Makefile:36: update-csv] Error 127
```

**Comment by rigazilla** [comment]
fixed the PATH setting. thanks @dmvolod

**Comment by dmvolod** [comment]
@rigazilla excellent, works fine now.
Please include regenerated `deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml` file in PR to prevent git change on first usage.

File: build/update-csv.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+OPERATOR_CSV_FILE="deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml"
+
+#Ensure that GOPATH/bin is in the PATH
+printf "Validating yq installation..."
+if ! [ -x "$(command -v yq)" ]; then
+  printf "Not found\n"
+  installYQ
+  PATH=$(go env GOPATH)/bin:$PATH
```
Comment (dmvolod) [inline build/update-csv.sh:24]:
```suggestion
```

File: build/update-csv.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+OPERATOR_ROLES_FILES=(
+    "deploy/role.yaml"
+    "deploy/clusterrole.yaml"
+)
+
+OPERATOR_CSV_FILE="deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml"
+
+#Ensure that GOPATH/bin is in the PATH
```
Comment (dmvolod) [inline build/update-csv.sh:19]:
```suggestion
PATH=$(go env GOPATH)/bin:$PATH
```

File: build/update-csv.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+YQ_MAJ_VERSION=${1:-4}
+installYQ() {
+    printf "Installing yq ..."
+    GO111MODULE=off go get -u github.com/myitcv/gobin
+    PATH=$(go env GOPATH)/bin:$PATH
```
Comment (dmvolod) [inline build/update-csv.sh:7]:
```suggestion
```

File: build/update-csv.sh (hunk 5 above)
Comment (rigazilla) [reply build/update-csv.sh:19]: @dmvolod I would change the user path only if yq is not available, i.e. move this inside the if statem

Comment (dmvolod) [reply build/update-csv.sh:18]: In this case how to validate `yq` in the PATH if PATH is not set?
For me script always install yq even if it already installed

Comment (rigazilla) [reply build/update-csv.sh:18]: if the user already have the correct yq in the path nothing is installed

Comment (dmvolod) [reply build/update-csv.sh:18]: For my laptop GOPATH/bin never installed in the PATH and `yq` always reinstalling on every codegen
It's better to set validate it, set GOPATH/bin into the PATH and validate again

File: build/update-csv.sh [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    printf "incorrect major version %s\n" "${INSTALLED_YQ_MAJ_VERSION}"
+    return 1
+  fi
+fi
+ 
+}
+
+
+#Ensure that GOPATH/bin is in the PATH
+printf "Validating yq installation..."
```
Comment (dmvolod) [inline build/update-csv.sh:39]:
```suggestion
```

File: build/update-csv.sh [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+if ! validateYQ; then
+  printf "Valid yq not found. Trying adding GOPATH/bin\n"
```
Comment (dmvolod) [inline build/update-csv.sh:41]:
```suggestion
  printf "Valid yq not found in PATH. Trying adding GOPATH/bin\n"
```

File: build/update-csv.sh [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    printf "%s\n" "${YQ_MAJ_VERSION}"
+  else
+    printf "incorrect major version %s\n" "${INSTALLED_YQ_MAJ_VERSION}"
+    return 1
+  fi
+fi
+ 
+}
+
+
```
Comment (dmvolod) [inline build/update-csv.sh:37]:
```suggestion
```

Comment (rigazilla) [reply build/update-csv.sh]: merged, thanks @dmvolod

**Review by dmvolod — APPROVED** [review]

**Comment by rigazilla** [comment]
merged, thanks @dmvolod for the review

**Pull request merged**